packages/palette/src/elements/HorizontalOverflow/HorizontalOverflow.tsx

```diff
@@ -0,0 +1,100 @@
+import { themeGet } from "@styled-system/theme-get"
+import React, { useEffect, useRef, useState } from "react"
+import styled from "styled-components"
+import {
+  compose,
+  justifyContent,
+  JustifyContentProps,
+  padding,
+  PaddingProps,
+} from "styled-system"
+import { visuallyDisableScrollbar } from "../.."
+import { splitProps } from "../../utils/splitProps"
+import { Box, BoxProps } from "../Box"
+
+const splitRailProps = splitProps<PaddingProps & JustifyContentProps>(
+  compose(padding, justifyContent)
+)
+
+const Overlay = styled(Box)<{ atEnd: boolean }>`
+  position: relative;
+
+  /* Fade-out gradient */
+  &::after {
+    display: block;
+    content: "";
+    position: absolute;
+    top: 0;
+    right: 0;
+    bottom: 0;
+    width: ${themeGet("space.6")};
+    z-index: 1;
+    pointer-events: none;
+    background: linear-gradient(
+      to right,
+      rgba(255, 255, 255, 0) 0%,
+      rgba(255, 255, 255, 1) 100%
+    );
+
+    /* Hide when scrolled all the way over */
+    transition: opacity 250ms;
+    opacity: ${({ atEnd }) => (atEnd ? 0 : 1)};
+  }
+`
+
+const Viewport = styled(Box)`
+  overflow-x: auto;
+  -webkit-overflow-scrolling: touch;
+  ${visuallyDisableScrollbar}
```

Comment:
I am visually disabling the scrollbar here now; since it's come up multiple times. I suspect we should maybe add the synthetic one at some point but this requires some design thought.